### PR TITLE
SAT: make `expect_records` mandatory in `high` `test_strictness_level`

### LINKED ISSUE #18222: SAT [strict_mode]: `basic_read.expect_records` must be set
## Tell us about the problem you're trying to solve
`basic_read.expect_record` is currently optionally used to declare a file with a list of records expected to receive on a `read` command. Making it mandatory for GA connectors is required to validate record schema and limit regression on `read`.

## Describe the solution you’d like
SAT must enforce the presence of the `expect_record` field in the `basic_read` test section for GA connectors.
I'd like these expect_records to be split by stream for maximum control and granularity in a future coverage computation.

**If a bypass_reason is filled SAT should log a warning.**

Example for a partially seeded sandbox account:
```yaml
strict_mode: strict
tests:
  basic_read:
    - config_path: "secrets/config.json"
      configured_catalog_path: "integration_tests/configured_catalog.json"
      expect_records:
        streams:
           - stream_name: stream_a
             expected_records_path: "integration_tests/a_expected_records.txt"
             extra_fields: no
             exact_order: yes
             extra_records: no
           - stream_name: stream_b
             bypass_reason: "We cannot seed stream b"
```

Example for a not seeded sandbox account:

```yaml
strict_mode: strict
tests:
  basic_read:
    - config_path: "secrets/config.json"
      configured_catalog_path: "integration_tests/configured_catalog.json"
      expect_records:
        bypass_reason: "No stream can be seeded on our sandbox account"
```

**If a bypass_reason is filled SAT should log a warning.**

This will require changing the configuration structure of `expect_records` which is not currently stream-specific:
```yaml
  basic_read:
    - config_path: "secrets/config.json"
      configured_catalog_path: "integration_tests/configured_catalog.json"
      expect_records:
        path: "integration_tests/expected_records.txt"
        extra_fields: no
        exact_order: yes
        extra_records: no
```

### PR DESCRIPTION
**Pull request by alafanechere**
## What
Closes https://github.com/airbytehq/airbyte/issues/18222
In `high` `test_strictness_level` we want to make the `expect_records` subtest mandatory and have the following guarantees:
* All the streams are present in the `expected_records.json` file
* If some streams are not present in the `expected_records.json`, they must be declared as `empty_streams` (with a valid `bypass_reason`)

## How
0. Refacto after https://github.com/airbytehq/airbyte/pull/18425:  Move the test enforcement for `high` strictness level in an `empty_streams` fixture used by test_read instead of enforcing it the test itself. Decoupling is better because multiple tests can use this fixture with the same enforced rules.
1. Improve `expected_records` fixture: make the tests using it fail if no `expect_records` subtest is set, or some streams are not seeded by comparing the current catalog to the streams existing in the `expected_records`. 

## Recommended reading order
1. https://github.com/airbytehq/airbyte/blob/1b5a5eb57e83a47ce887baa20e339ce286d72995/airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/conftest.py#L172
2. https://github.com/airbytehq/airbyte/blob/1b5a5eb57e83a47ce887baa20e339ce286d72995/airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py#L470
3. https://github.com/airbytehq/airbyte/blob/1b5a5eb57e83a47ce887baa20e339ce286d72995/airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/config.py#L76

## 🚨 User Impact 🚨
No configuration structure change: existing `acceptance-test-config.yml` should still be supported. I'll test it below with `/test`

### CODE REVIEW

**Comment by alafanechere** [comment]
☝️ The failure above was expected: 
* I migrated `source-google-analytics-v4`'s `acceptance-test-config.yml` and set `test_strictness_level: high`
* The catalog used for the basic read test contains more streams than the one declared in `expected_records.json` -> Failure
```
 Failed: High strictness level error: two_weekly_active_users, four_weekly_active_users, monthly_active_users, traffic_sources, website_overview streams are declared in the catalog but do not have expected records. Please add expected records to integration_tests/expected_records.txt or declare these streams in empty_streams.
 ```

**Comment by alafanechere** [comment]
☝️ This test should succeed, empty streams  detected in the previous test run were declared with `bypass_reasons`.

**Comment by alafanechere** [comment]
☝️ I reverted `source-google-analytics-v4`'s `acceptance-test-config` to its original format. Tests should succeed.

File: airbyte-integrations/bases/source-acceptance-test/.coveragerc [hunk 1]
```diff
@@ -1,8 +1,8 @@
 [report]
 # show lines missing coverage
 show_missing = true
-# coverage 64% measured on 62303a85def89450d2e46573a3d96cd326f2e921 (2022-08-09)
+# coverage 74% measured on 4977ac2c527f03c15ce0094cfd48f6104a0fd82f (2022-10-26)
 # This number should probably never be adjusted down, only up i.e: we should only ever increase our test coverage
-fail_under = 64
+fail_under = 74
```
Comment (alafanechere) [inline airbyte-integrations/bases/source-acceptance-test/.coveragerc:6]: I took advantage of this PR to increase the coverage threshold

**Review by evantahler — APPROVED** [review]

**Comment by alafanechere** [comment]
/publish connector=bases/source-acceptance-test auto-bump-version=false

**Pull request merged**